olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -832,9 +833,17 @@ DefaultCache::StorageOpenResult DefaultCacheImpl::SetupProtectedCache() {
   StorageSettings protected_storage_settings;
   protected_storage_settings.max_file_size = 32 * 1024 * 1024;
 
-  auto status = protected_cache_->Open(
-      settings_.disk_path_protected.get(), settings_.disk_path_protected.get(),
-      protected_storage_settings, OpenOptions::ReadOnly);
+  OpenOptions open_mode = settings_.openOptions;
+  if (!(open_mode & OpenOptions::ReadOnly)) {
+    if (utils::Dir::IsReadOnly(settings_.disk_path_protected.get() +
+                               "\\LOCK")) {
+      open_mode = static_cast<OpenOptions>(open_mode | OpenOptions::ReadOnly);
+    }
```

Comment:
Add info log here stating that we need to open the protected cache in r/o mode, to avoid surprises when there is a crash somewhere.